sympy/solvers/ode/tests/test_single.py

```diff
@@ -377,6 +377,18 @@ def test_2nd_2F1_hypergeometric_intrgral():
     assert checkodesol(eq, sol) == (True, 0)
 
 
+def test_2nd_nonlinear_autonomous_conserved():
+    _ode_solver_test(_get_examples_ode_sol_2nd_nonlinear_autonomous_conserved)
+
+
+def test_2nd_nonlinear_autonomous_conserved_integral():
+    eq = f(x).diff(x, 2) + asin(f(x))
+    sol = [Eq(Integral(1/sqrt(C1 - 2*Integral(asin(u), u)), (u, 1, f(x))), C2 + x),
+    Eq(Integral(1/sqrt(C1 - 2*Integral(asin(u), u)), (u, 1, f(x))), C2 - x)]
+    assert sol == dsolve(eq, hint='2nd_nonlinear_autonomous_conserved_Integral', simplify=False)
+    assert checkodesol(eq, [s.doit() for s in sol]) == [(True, 0), (True, 0)]
```

Comment:
Why is the doit here needed. We should improve `checkodesol` to handle this case.

Reply:
I'll look into it.